[17.0][FW] account_financial_report: multiple ports from 14.0

henrybackman opened the pull request:
Port of the following PRs from 14.0 to 17.0:
- #932
- #1154
- #1166

jdidderen committed:
[FIX] account_financial_report - fix abstract report class name

toita86 committed:
[IMP] account_financial_report: localize date format on VAT report

Before this change, the date format in the VAT report print was not sensible to the company's localization.
The date string has been converted to a Date field to ensure the correct format aligns with the company's localization.

WesleyOliveira98 committed:
[IMP] account_financial_report: general ledger filter by account types

toita86 committed:
[IMP] account_financial_report: VAT report fields translation

mmequignon reviewed (approved):
Not much context, but code looks good

pedrobaeza commented:
Well, the CI is red, so first, that should be fixed.

mmequignon commented:
> Well, the CI is red, so first, that should be fixed.

You're right, I missed this, `user_type_id` has been replaced by `account_type` (Many2one to Selection)

henrybackman committed:
[FIX] account_financial_report: fix account_type field

When porting from v14 to v17, the `account_account.user_type_id`
has been changed to `account_type`. Previously `general_ledger_wizard`
was able to store multiple account types through the `account_type_ids`
field. To simplify the forward porting, for now only one `account_type`
can be selected in the wizard.

henrybackman commented:
> > Well, the CI is red, so first, that should be fixed.
> 
> You're right, I missed this, `user_type_id` has been replaced by `account_type` (Many2one to Selection)

Yes, this was the existing issue and it has now been resolved by changing the field in General Ledger Wizard to `account_type` as well. This however means that the wizard supports only one account type instead of many, but it was considered that expanding to multiple was beyond the scope of the forward porting.

henrybackman commented:
@WesleyOliveira98 While forward porting your changes, I made a change to the General Ledge Wizard to accommodate changes in 17.0. This has a functional impact:
- Before it was possible to select multiple account types but I implemented a simplified fix which allows selecting only a single account type 

Could I ask you to review the port and my changes?
Fix commit:
- [https://github.com/OCA/account-financial-reporting/pull/1233/commits/f980e17eb386906c228cad553d3eaa6c8ccee649](https://github.com/OCA/account-financial-reporting/pull/1233/commits/f980e17eb386906c228cad553d3eaa6c8ccee649)

WesleyOliveira98 commented:
> * Before it was possible to select multiple account types but I implemented a simplified fix which allows selecting only a single account type

Hello @henrybackman, I don't think this is a good change, users sometimes select more than one account type at the same time
